src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -36,9 +36,9 @@ void CoreWorkerDirectTaskSubmitter::AddWorkerLeaseClient(
                                   std::make_pair(std::move(lease_client), expiration));
 }
 
-void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(const rpc::WorkerAddress &addr,
-                                                 const SchedulingKey &scheduling_key,
-                                                 bool was_error) {
+void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(
+    const rpc::WorkerAddress &addr, const SchedulingKey &scheduling_key, bool was_error,
+    const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> &assigned_resources) {
```

Comment:
should we return the resources back to raylet when we return the worker to it?  Seems like the current code doesn't handle it.

Reply:
There's no need, this is just for get_resource_ids(). The tracking is separate.